### PR TITLE
DX: add proper test for `SelfAccessorFixer` must run before `SelfAccessorFixer`

### PR DESCRIPTION
**Pull request by kubawerlos**

### CODE REVIEW

**Review by Wirone — APPROVED** [review]

**Comment by keradus** [comment]
> test for `SelfAccessorFixer` must run before `SelfAccessorFixer`

😅

File: tests/IntegrationTest.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ protected static function getFixturesDir(): string
 
     protected static function getTempFile(): string
     {
-        return tempnam(sys_get_temp_dir(), 'PHP-CS-Fixer');
+        return sys_get_temp_dir().\DIRECTORY_SEPARATOR.'MyClass.php';
```
Comment (keradus) [inline tests/IntegrationTest.php:40]: why the change? it could be problematic when running different integration test in parallel

Comment (kubawerlos) [reply tests/IntegrationTest.php:40]: How could it be problematic? Aren't the tests run in parallel already so it would crash already?

Also, isn't it a good practice not to have tests non-deterministic? If there is a need of having a different path for each test better would be to have a deterministic way to do it.

Comment (keradus) [reply tests/IntegrationTest.php:40]: I do not have good understanding of newest PHPUnit parallelization and I imagine that now/later it could make parallel test cases execution, not only test suites - in that case, if all test cases would run in parallel, they would fight and override same test file - that's why the test file was always different in original code.

would you mind to share the reasoning?
> why the change?

Comment (kubawerlos) [reply tests/IntegrationTest.php:40]: Fixer `psr_autoloading` needs a file name to get a class name, with a name like `/tmp/PHP-CS-FixernJXNlg` that would not be possible: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.21.2/src/Fixer/Basic/PsrAutoloadingFixer.php#L106

Comment (keradus) [reply tests/IntegrationTest.php:40]: thanks for reasoning, now it makes more sense!
